src/objects/Object.h

```diff
@@ -0,0 +1,30 @@
+#ifndef OBJECT_H
+#define OBJECT_H
+
+#include "stdlib.h"
+#include <iostream>
+
+class Level;
+
+class Object {
+private:
+    Level* level;
+
+public:
+    int64_t objectUID;    
+    bool shouldUpdate = true;
+
+public:
+    ~Object() { destroyed(); }
+
+public:
+    virtual void spawned() { }
+    virtual void update(float delta) { }
+    virtual void destroyed() { }
+
+public:
+    Level* getLevel() { return level; }
+    void setLevel(Level* lvl) { level = lvl; }
```

Comment:
Ты уверен, что уровень может измениться в рантайме? Я бы предложил удалить этот метод, а поле level инициализировать в конструкторе
```suggestion
    void Object(Level& lvl) : level(lvl) {}
```